mochify/lib/load-config.js

```diff
@@ -10,19 +10,42 @@ async function loadConfig(options) {
 
   const default_config_promise = explorer.search();
 
+  let merged;
   if (options.config) {
     const specified = await explorer.load(options.config);
     const config = Object.assign(specified.config, options);
-    return mergeWithDefault(default_config_promise, config);
+    merged = await mergeWithDefault(default_config_promise, config);
+  } else {
+    merged = await mergeWithDefault(default_config_promise, options);
   }
 
-  return mergeWithDefault(default_config_promise, options);
+  const validation_error = validate(merged);
+  if (validation_error) {
+    throw validation_error;
+  }
+  return merged;
 }
 
 async function mergeWithDefault(default_config_promise, config) {
   const default_config = await default_config_promise;
   if (default_config) {
-    return deepmerge(default_config.config, config);
+    return deepmerge(default_config.config, config, {
+      clone: false
```

Comment:
Cloning a stream apparently causes infinite recursion. Don't think not cloning will hurt us here.